web/src/navigator.rs

```diff
@@ -31,10 +31,13 @@ impl WebNavigatorBackend {
         let window = web_sys::window().expect("window()");
         let performance = window.performance().expect("window.performance()");
 
-        // Upgarde to HTTPS takes effect if the current page is hosted on HTTPS.
+        // Upgrade to HTTPS takes effect if the current page is hosted on HTTPS.
         let upgrade_to_https =
             upgrade_to_https && window.location().protocol().unwrap_or_default() == "https:";
 
+        // Adding trailing slash so url::parse will not drop last part
+        base_url = base_url.map(|x| if x.ends_with('/') { x } else { x + "/" });
+
```

Comment:
We can put this inside of the `if` statement directly below to avoid the `map`

Reply:
done. I totally understand everything about rust and everything I just did and I didn't just change stuff according to compiler suggestions until it compiled.